core/globals.h

```diff
@@ -912,7 +912,15 @@ struct _dcontext_t {
 
     dr_where_am_i_t whereami; /* where control is at the moment */
 #ifdef UNIX
+/* On ARM based machines, char is unsigned by default.
```

Comment:
Yeah, without this `signals_pending` wraps and becomes 255, instead of -1. This causes issues with the `cmp` `le` check in `append_fcache_enter_prologue`.

Reply:
For completeness of this comment thread:
it seems that making the `char` `signed` didn't help with the assembly `cmp le` check (failures in `linux.signal0111`). 

But it is still required for correctness of checks in C code. Note that when I made the `char` `signed`, it did fix `linux.signest`.

More details in other comment threads in the PR.